### PR TITLE
MRG fix bug with stim_artifact fixer

### PR DESCRIPTION
**Pull request by mainakjas**
cc @agramfort I'll let you check and merge since you wrote this

closes https://github.com/mne-tools/mne-python/issues/2385

### CODE REVIEW

**Comment by mainakjas** [comment]
... hold on, test fails. Fixing it ...

**Comment by mainakjas** [comment]
@agramfort any idea what this test is supposed to achieve: https://github.com/mne-tools/mne-python/blob/master/mne/preprocessing/tests/test_stim.py#L70 ? in fact, how are we testing if we have no data with stim artifacts? I don't see any simulation either.

**Comment by joewalter** [comment]
@mainakjas thanks for working on this problem. Can you please check, that all channels are corrected. I was recently testing the new function and basically all epochs were rejected while extracting epochs from the stim-corrected raw as the EOG channel was not corrected and thus the rejection threshold was reached for each trial. With my old function, which was replaced, this was not the case. So there were obviously some undesired changes of the properties of the stim-artifact correction.

**Comment by mainakjas** [comment]
sure I can take a look @joewalter. Can you explain the logic behind the tests? (I realized that you are the author of the line in question, not alex). Why `assert_true(np.all(data) == 0.)` and `assert_array_almost_equal(diff_data0, np.zeros(len(diff_data0)))` ?

**Comment by mainakjas** [comment]
btw, I will / might be AFK for the next few days. Do you want to take over and see for yourself what's going on?

File: mne/preprocessing/tests/test_stim.py [hunk 1]
```diff
@@ -51,9 +51,6 @@ def test_fix_stim_artifact():
 
     # use window before stimulus in raw
     event_idx = np.where(events[:, 2] == 1)[0][0]
-    tmin, tmax = -0.045, -0.015
-    tmin_samp = int(-0.035 * raw.info['sfreq'])
-    tmax_samp = int(-0.015 * raw.info['sfreq'])
```
Comment (agramfort) [inline mne/preprocessing/tests/test_stim.py]: why these deletions?

File: mne/preprocessing/tests/test_stim.py [hunk 2]
```diff
@@ -73,9 +70,6 @@ def test_fix_stim_artifact():
     tmin, tmax, event_id = -0.2, 0.5, 1
     epochs = Epochs(raw, events, event_id, tmin, tmax, picks=picks,
                     preload=True, reject=None)
-    e_start = int(np.ceil(epochs.info['sfreq'] * epochs.tmin))
-    tmin_samp = int(-0.035 * epochs.info['sfreq']) - e_start
-    tmax_samp = int(-0.015 * epochs.info['sfreq']) - e_start
```
Comment (agramfort) [inline mne/preprocessing/tests/test_stim.py]: why these deletions?

**Comment by agramfort** [comment]
assert_true(np.all(data) == 0.) 

makes sure if the interval used for artifact it the full epoch length then it's all zeros. Might not work for Raw though...

assert_array_almost_equal(diff_data0, np.zeros(len(diff_data0))) ?

test that the values in between are constants?

**Comment by agramfort** [comment]
travis is not happy as you certainly noticed :)

File: mne/preprocessing/tests/test_stim.py (hunk 1 above)
Comment (mainakjas) [inline mne/preprocessing/tests/test_stim.py]: they are already defined. I don't know why they were redefined here -- it seemed redundant to me.

Comment (mainakjas) [inline mne/preprocessing/tests/test_stim.py]: ... oh, I think they are different. I'll check and put them back.

Comment (agramfort) [inline mne/preprocessing/tests/test_stim.py]: yeah I think they are different.

let me know when travis is happy.

**Comment by agramfort** [comment]
what's our status here?

**Comment by mainakjas** [comment]
I'll finish this after 12th ...

**Comment by mainakjas** [comment]
ok, test fixed I think. Travis should be happy now.

**Comment by agramfort** [comment]
Thanks

**Pull request merged**